ENIGMAsystem/SHELL/Graphics_Systems/General/GSfont.cpp

```diff
@@ -133,7 +107,7 @@ namespace enigma_user {
 double string_char_width(variant vstr)
 {
   string str = toString(vstr);
-  get_font(fnt,currentfont,0);
+  SpriteFont& fnt = sprite_fonts[currentfont];
```

Comment:
This would default construct a font for -2 and everything else and not just the default font. This completely destroys the ability to see if a font actually exists. I'm not sure the macros were correct either though, I believe my original intention was that AssetArray::get was supposed to perform bounds checking like STL does, and all asset functions like this would be required to get the resource that way.